base/array.jl

```diff
@@ -827,6 +836,18 @@ function deleteat!(a::Vector, inds)
     return a
 end
 
+function deleteat!(a::Vector, inds::AbstractVector{Bool})
```

Comment:
Ah, shucks, right of course.  That's unfortunate.  Yup, knowing that the logical index is sorted is extremely valuable information.  And maybe someday we'll extend indexing support for all iterables.  I'm not sure that `to_indices` will gain us much when it's restricted to only affecting arrays in the first place… but we may as well allow this.